File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    const answer = [1];
    let days = Math.ceil((100 - progresses[0]) / speeds[0])
    
    for (let i = 1; i < progresses.length; i++) {
        const progress = progresses[i], speed = speeds[i]
        
        if (progress + speed * days >= 100) {
            answer[answer.length - 1] += 1
        } else {
            answer.push(1)
            days = Math.ceil((100 - progress) / speed)
        }
    }
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
from math import ceil

def solution(progresses, speeds):
    answer = [1]
    days = ceil((100 - progresses[0]) / speeds[0])
    
    for progress, speed in zip(progresses[1:], speeds[1:]):
        if progress + speed * days >= 100:
            answer[-1] += 1
        else:
            answer.append(1)
            days = ceil(100 - progress) / speed
    
    return answer